pinot-core/src/main/java/org/apache/pinot/core/operator/query/AggregationGroupByOrderByOperator.java

```diff
@@ -145,4 +148,19 @@ public ExecutionStatistics getExecutionStatistics() {
     return new ExecutionStatistics(_numDocsScanned, numEntriesScannedInFilter, numEntriesScannedPostFilter,
         _numTotalDocs);
   }
+
+  private int calculateMinTrimSize () {
+    // In query option, if a positive min trim size is given, we use it to override the server settings. Otherwise
+    // check if a simple boolean option is given and use default trim size.
+    QueryOptions queryOptions =  new QueryOptions(_queryContext.getQueryOptions());
```

Comment:
Use the static method instead of the constructor to avoid the overhead of parsing unrelated fields